### DISCUSSION TITLE
Pug

### DISCUSSION THREAD

**Discussion by louiss0** [withastro/roadmap#229] [Proposal]
I want to have the power to write pug code inside of `.astro` files can you please do that for me. I like pug it makes writing HTML simple in most ways than other templating engines. There is a vite plugin for pug but that sucks.

**Comment by nanarino** [comment]
I am a beginner with Astro. I have a css library project that uses Stylus variables to concatenate class name prefixes. The documentation for the styles uses Pug to insert the prefix and the marked filter. Now, I want to migrate the project to Astro and it would be easy if Pug is supported.

**Comment by ar-rm** [comment]
Originally mentioned here in October 2021:

[RFC: Add first-class support for Pug in .astro files #1513](https://github.com/withastro/astro/issues/1513)
